src/routing_table/mod.rs

```diff
@@ -1129,7 +1129,8 @@ mod tests {
         assert_eq!(our_name, table.our_name);
 
         // Get some names
-        let close_name: u32 = *unwrap!(table.our_section.iter().nth(4));
+        let close_name: u32 =
+            *unwrap!(table.our_section.iter().filter(|name| **name != our_name).nth(4));
```

Comment:
This unwrap looks dangerous in production code and not in the allowed use cases (according to our standards so far). I wonder if perhaps a match and at least print an `error!` if it fires is safer?